### PR TITLE
Cart Transform template for JavaScript and TypeScript

### PR DESCRIPTION
**Pull request by nickwesselman**
Implements a basic JavaScript and TypeScript template for the Cart Transform API.

## 🎩 
```
npm run generate extension -- --type cart_transform --clone-url https://github.com/Shopify/function-examples#cart-transform-js-template
```

### CODE REVIEW

File: checkout/javascript/cart-transform/default/schema.graphql [hunk 1]
```diff
@@ -0,0 +1,3111 @@
...
+  """
+  The tax amount for the customer to pay at checkout.
+  """
+  totalTaxAmount: MoneyV2
+}
+
+"""
+Information about the options available for one or more line items to be delivered to a specific address.
+"""
+type CartDeliveryGroup {
```
Comment (ignacio-chiazzo) [inline checkout/javascript/cart-transform/default/schema.graphql:119]: This schema is outdated it should only have `cartLines` cc @siddhantbajaj

**Review by ignacio-chiazzo — COMMENTED** [review]
Thanks @nickwesselman!!!

File: checkout/javascript/cart-transform/default/schema.graphql (hunk 1 above)
Comment (nickwesselman) [reply checkout/javascript/cart-transform/default/schema.graphql:119]: I pulled this from your Rust template, is it out of date as well?

Comment (nickwesselman) [reply checkout/javascript/cart-transform/default/schema.graphql]: I pulled and updated the schema for Rust and JS

File: checkout/rust/cart-transform/default/src/main.rs [hunk 2]
```diff
@@ -14,7 +14,7 @@ generate_types!(
 #[shopify_function]
 fn function(_input: input::ResponseData) -> Result<output::FunctionResult> {
     let no_changes = output::FunctionResult {
-        operations: Some(vec![]),
+        operations: vec![],
```
Comment (ignacio-chiazzo) [inline checkout/rust/cart-transform/default/src/main.rs:17]: https://github.com/Shopify/shopify/commit/d322403ec033b68a55dbc61aa46c75b8d2b8522a

**Review by ignacio-chiazzo — APPROVED** [review]
:shipit: 

Screenshot 2023-06-26 at 4 39 46 PM

File: checkout/javascript/cart-transform/default/src/index.liquid [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+export default /**
+ * @param {InputQuery} input
+ * @returns {FunctionResult}
+ */
+(input) => {
+  return NO_CHANGES;
+};
+{%- elsif flavor contains "typescript" -%}
+import {
```
Comment (kmdavis) [inline checkout/javascript/cart-transform/default/src/index.liquid:24]: this should use `import type`

```suggestion
import type {
```

**Review by kmdavis — APPROVED** [review]
one nit, but otherwise, the JS/TS looks good

**Comment by ignacio-chiazzo** [comment]
Tested and worked!

**Pull request merged**